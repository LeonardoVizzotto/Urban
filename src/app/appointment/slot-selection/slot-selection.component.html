<app-container *ngIf="slots$ | async; let slots">
  <span>Please chose the your preffered time period to schedule a session</span>
  <app-card
    *ngFor="let slot of slots | slice: 0:count"
    (onClick)="selectSlot(slot)"
  >
    {{ slot.localisedTime }} - {{ slot.price }}
  </app-card>
  <app-button
    *ngIf="slots.length > count"
    [appearance]="'secondary-ghost'"
    (onClick)="loadMoreSlots()"
    >Load more slots</app-button
  >
</app-container>
