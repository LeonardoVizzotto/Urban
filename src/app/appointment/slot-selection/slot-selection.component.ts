import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { ROUTES } from 'src/app/routes.constants';
import { BasketService } from 'src/app/services/basket/basket.service';
import { Slot } from 'src/app/services/slots/model';
import { SlotsService } from 'src/app/services/slots/slots.service';

const SLOT_INCREMENT = 5;

@Component({
  selector: 'app-slot-selection',
  templateUrl: './slot-selection.component.html',
  styleUrls: ['./slot-selection.component.scss'],
})
export class SlotSelectionComponent {
  public count = SLOT_INCREMENT;
  public slots$ = this.slotsService.getSlots();

  constructor(
    private slotsService: SlotsService,
    private basketService: BasketService,
    private router: Router,
  ) {}

  public loadMoreSlots() {
    this.count += SLOT_INCREMENT;
  }

  public selectSlot(slot: Slot) {
    this.basketService.selectSlot(slot);
    this.router.navigate([ROUTES.WORKERS]);
  }
}
