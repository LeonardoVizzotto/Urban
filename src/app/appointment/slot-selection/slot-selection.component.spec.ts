import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SlotSelectionComponent } from './slot-selection.component';

describe('SlotSelectionComponent', () => {
  let component: SlotSelectionComponent;
  let fixture: ComponentFixture<SlotSelectionComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ SlotSelectionComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SlotSelectionComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
