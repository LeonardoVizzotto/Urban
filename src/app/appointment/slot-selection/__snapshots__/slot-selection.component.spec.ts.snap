// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotSelectionComponent should match snapshot 1`] = `
<app-slot-selection
  basketService={[Function BasketService]}
  count={[Function Number]}
  router={[Function Object]}
  slots$={[Function Observable]}
  slotsService={[Function Object]}
>
  <app-container>
    <app-card>
      <button
        class="card"
      >
         12:00 - £81.00 
      </button>
    </app-card>
    <app-card>
      <button
        class="card"
      >
         12:30 - £81.00 
      </button>
    </app-card>
    <app-card>
      <button
        class="card"
      >
         13:00 - £87.00 
      </button>
    </app-card>
  </app-container>
</app-slot-selection>
`;
